esp/src/src/WsTopology.ts

```diff
@@ -76,6 +78,11 @@ const TpLogFileStore = declare([Memory, Evented], {
                     deferredResults.total.resolve(data.length);
                 }
                 return deferredResults.resolve(this.data);
+            })).then(lang.hitch(this, function (response) {
```

Comment:
Was this `then` the actual fix? Any idea what would have changed to require this?

Reply:
This code simply disables / enables the download buttons until the log data is loaded (can be slow).